23_Arithmetic2UnitTest/main.cpp

```diff
@@ -394,47 +456,64 @@ class ArithmeticUnitTestApp final : public application_templates::BasicMultiQueu
 		const auto testData = reinterpret_cast<const type_t*>(dataFromBuffer + 1);
 		// TODO: parallel for (the temporary values need to be threadlocal or what?)
 		// now check if the data obtained has valid values
-		type_t* tmp = new type_t[itemsPerWG];
-		type_t* ballotInput = new type_t[itemsPerWG];
+		type_t* tmp;
+		if constexpr (WorkgroupTest)
+			tmp = new type_t[itemsPerWG];
+		else
+			tmp = new type_t[itemsPerWG * itemsPerInvoc];
 		for (uint32_t workgroupID = 0u; success && workgroupID < workgroupCount; workgroupID++)
 		{
-			const auto workgroupOffset = workgroupID * itemsPerWG;
-
 			if constexpr (WorkgroupTest)
 			{
-				if constexpr (std::is_same_v<ballot<type_t>, Binop>)
+				const auto workgroupOffset = workgroupID * itemsPerWG;
+				Arithmetic<Binop>::impl(tmp, inputData + workgroupOffset, itemsPerWG);
```

Comment:
why is `itemsPerInvoc` only taken into account for subgroups?

its quite confusing to have `itemsPerInvoc` when `itemsPerWG` is supposed to be the number of inputs each workgroup consumes (and a subgroup is the workgroup for subgroup only tests)